extensions/flyway/deployment/src/test/resources/baseline-on-migrate.properties

```diff
@@ -1,7 +1,7 @@
-quarkus.datasource.url=jdbc:h2:tcp://localhost/mem:test-quarkus-baseline-on-migrate;DB_CLOSE_DELAY=-1;INIT=RUNSCRIPT FROM 'src/test/resources/h2-init-data.sql'
-quarkus.datasource.driver=org.h2.Driver
+quarkus.datasource.db-kind=h2
```

Comment:
Can we make `db-kind` optional for cases where there is only one DB kind enabled in the pom.xml?
This will be the case for probably >80% of apps, and it would reduce the number of required config options from 4 to 3.

Reply:
Btw, my suggestion is not a hard requirement for this PR. The `db-kind` property could always be made optional later.

Reply:
So the thing is that it's the only mandatory build time property so basically the only thing saying you have a datasource that is available at build time.

So for now, no.

Reply:
Well we have the fact that a driver extension is present (and only one) via possible BuildItem publications. It's equally build time. But that shoudl have its own GitHub issue.

Reply:
You know you have a driver extension but you don't know a datasource has been configured if all the datasource configuration is runtime as you shouldn't access the runtime config at build time.

We could try to assert some runtime configuration is here but that's very fragile. That's the main issue with doing this.

If we decide to do that, that would mean move more things at runtime as we wouldn't be able to know if a datasource is configured at build time.

Reply:
Ok right, that's the core of the problem we can discuss on the ML but after this PR. 
Is it that bad to read runtime properties at build time and assume that some of its content is frozen at build time. In the JDBC url case, that would mean `jdbc:db-kind`